src/components/app.tsx

```diff
@@ -5,13 +5,43 @@ import { DragDropContext } from 'react-dnd';
 import HTML5Backend from 'react-dnd-html5-backend';
 import * as SplitPane from 'react-split-pane';
 
+import {Data, InlineData, isInlineData, isUrlData, UrlData} from 'vega-lite/build/src/data';
+
 import {DataPane} from './data-pane';
 import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
+import {
+  ActionHandler,
+  DatasetAsyncAction,
+  datasetUrlLoad,
+} from '../actions';
+
+
+export type VoyagerConfig = Object;
+export type VoyagerData = InlineData | UrlData;
+
+interface VoyagerAppProps extends ActionHandler<DatasetAsyncAction> {
+  config: Object;
+  data: Data;
+  dispatch: any;
+}
+
+class AppBase extends React.PureComponent<VoyagerAppProps, {}> {
+
+  constructor(props: any) {
```

Comment:
@tafsiri  I tried changing `any` to `VoyagerAppProps` and this throws a reasonable error.  

Basically `VoyagerAppProps extends ActionHandler<DatasetAsyncAction>` but your call of App in both index and lib-voyager does not have the right property required by `ActionHandler<DatasetAsyncAction>`.   

I haven't thought much about whether you need to extend `ActionHandler<DatasetAsyncAction>`here, but I have to take back what I say yesterday.  I think the error from any here is a different problem from what I see error.  So I'd try to remove any and add the right type and try to correct things to match.